### PR TITLE
Serialize and deserialize program

### PR DESCRIPTION
**Pull request by pefontana**
# Add a Serialize and Deserialize method for the Program struct

## Description

* Create the ProgramSerializer Struct used to Serialize and Deserialize a Program struct. The goal of this struct, is to be equal to the ProgramJson but is to keep the Default Serialization and Deserialization traits 
* Add the method to Serialize and Deserialize a Program struct

## Checklist
- [x] Linked to Github Issue
- [ ] Unit tests added
- [x] Integration tests added.
- [x] This change requires new documentation.
  - [ ] Documentation has been added/updated.
  - [x] CHANGELOG has been updated.

### CODE REVIEW

File: vm/src/serde/deserialize_program.rs [hunk 1]
```diff
@@ -415,6 +412,22 @@ pub fn deserialize_felt_hex<'de, D: Deserializer<'de>>(d: D) -> Result<Felt252,
...
 }
 
+pub fn serialize_program_data<S: Serializer>(
+    v: &[MaybeRelocatable],
+    serializer: S,
+) -> Result<S::Ok, S::Error> {
+    let v = v
+        .iter()
+        .map(|val| match val {
+            MaybeRelocatable::Int(value) => format!("0x{:x}", value.to_biguint()),
```
Comment (fmoletta) [inline vm/src/serde/deserialize_program.rs:422]:
```suggestion
            MaybeRelocatable::Int(value) => format!("{:#x}", value.to_biguint()),
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,8 +2,11 @@
 
 #### Upcoming Changes
 
+* feat: Implement the Serialize and Deserialize methods fot the Program struct [#1458](https://github.com/lambdaclass/cairo-vm/pull/1458)
```
Comment (Oppen) [inline CHANGELOG.md:5]:
```suggestion
* feat: Implement the Serialize and Deserialize methods for the Program struct [#1458](https://github.com/lambdaclass/cairo-vm/pull/1458)
```

File: vm/src/serde/deserialize_program.rs [hunk 3]
```diff
@@ -70,7 +70,7 @@ impl BuiltinName {
 }
 
 #[cfg_attr(all(feature = "arbitrary", feature = "std"), derive(Arbitrary, Clone))]
-#[derive(Deserialize, Debug)]
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
```
Comment (fmoletta) [inline vm/src/serde/deserialize_program.rs:73]: Isn't the goal of the `ProgramSerializer` to remove the serialization responsibility from `ProgramJson`?

Comment (pefontana) [reply vm/src/serde/deserialize_program.rs]: Nice catch @fmoletta !
Yes, i removed the Serialize implementation of the ProgramJson

**Comment by pefontana** [comment]
> > The goal of this struct, is to be equal to the ProgramJson but is to keep the Default Serialization and Deserialization traits
> 
> Why can't `ProgramJson` derive them?

Because the ProgramJson has custom implementations of the Deserialize trait in some fields,
That causes errors when Serializing(with the default implementation) and the Deserilizing the ProgramJson struct

**Review by Oppen — APPROVED** [review]

**Review by fmoletta — APPROVED** [review]

**Pull request merged**